frontend/src/metabase/search/util/filter-types.ts

```diff
@@ -0,0 +1,23 @@
+import { FC } from "react";
+import { SearchModelType } from "metabase-types/api";
+
+export const SearchFilterKeys = {
+  Type: "type",
+} as const;
+
+export type TypeFilterProps = SearchModelType[];
+
+type SearchFilterPropTypes = {
```

Comment:
This is set up in a way where if we add, for example, a `created_by` key in `SearchFilterKeys`, the type-checker will complain that we don't have props set up for that specific filter key. This basically means that typescript will make sure that we have props and a component for each filter that we want to add to the search filter modal